integration-tests/startup.spec.js

```diff
@@ -0,0 +1,117 @@
+'use strict'
+
+const {
+  FakeAgent,
+  spawnProc,
+  curlAndAssertMessage
+} = require('./helpers')
+const path = require('path')
+const assert = require('assert')
+
+const startupTestFile = path.join(__dirname, 'startup/index.js')
+
+describe('startup', () => {
+  let agent
+  let proc
+
+  context('programmatic', () => {
+    beforeEach(async () => {
+      agent = new FakeAgent()
+    })
+
+    afterEach(() => {
+      proc.kill()
+      agent.close()
+    })
+
+    it('works for options.port', async () => {
+      proc = await spawnProc(startupTestFile, {
+        env: {
+          AGENT_PORT: agent.port
+        }
+      })
+      return curlAndAssertMessage(agent, proc, ({ headers, payload }) => {
+        assert.strictEqual(headers.host, `127.0.0.1:${agent.port}`)
```

Comment:
`assert` does not give meaningful error messages.

Reply:
I don't believe this is true.

Here is a quick `mocha` test I whipped up to demonstrate:

```js
'use strict'

const { expect } = require('chai')
const assert = require('assert')

const headers = {
  host: 'incorrect-host:4321'
}

describe('foo', () => {
  it('failing expect test', () => {
    expect(headers.host).to.equal('127.0.0.1:1234')
  })
  it('failing assert test', () => {
    assert.strictEqual(headers.host, '127.0.0.1:1234')
  })
})
```

The output I see from this on Node.js 14 looks like this:

<img width="690" alt="Screen Shot 2021-01-21 at 11 51 26 AM" src="https://user-images.githubusercontent.com/110455/105383518-33cf7280-5bdf-11eb-90fe-b8492e8de734.png">

It really seems to me like `assert` gives meaningful error messages.

Reply:
`chai` also allows more explicit checks like `expect(headers).to.have.property('host', '127.0.0.1:1234')` which gives more information about the actual underlying issue. I use this a lot especially when fixing multiple tests simultaneously so that I don't have to constantly go back to the line to remember what the test is doing.

Reply:
Switched to chai/assert.